src/api/routes/userRouter.js

```diff
@@ -2,11 +2,12 @@ import { Router } from 'express';
 import ProfilesController from '../controllers/profiles';
 import Auth from '../../middleware/auth';
 import validateBody from '../../middleware/validateBody';
+import check from '../../middleware/checkOwner';
 
 const userRouter = Router();
 const { updateProfile } = ProfilesController;
 const { verifyToken } = Auth;
 
-userRouter.put('/', verifyToken, validateBody('updateUser'), updateProfile);
+userRouter.put('/', verifyToken, check.profileOwner, validateBody('updateUser'), updateProfile);
```

Comment:
This middleware is not necessary as token verification ensures the user exists. What can be done is to update the `404` status code in the controller to `403`.